docs/5_2_hybrid_co-simulation_api.adoc

```diff
@@ -34,24 +34,28 @@ The Co-Simulation master can also call <<fmi3EnterEventMode>> at communication i
 If an FMU provides the early-return capability that includes the handling of events in *Event Mode*,
 the FMU signals this via <<canReturnEarlyAfterIntermediateUpdate>> in the <<modelDescription.xml>>.
 
-The FMU stops computation at the first encountered internal event (if any) and the event time is provided through <<intermediateUpdateTime>>.
-If <<fmi3DoStep>> returns with <<earlyReturn,`earlyReturn == fmi3True`>> and an event has happened, i.e. <<eventOccurred,`eventOccurred == fmi3True`>>, then an event handling has to be done by the Co-Simulation master.
-In order to start event handling the Co-Simulation master has to call <<fmi3EnterEventMode>> for that FMU to push the FMU into *Event Mode*.
+The FMU stops computation at the first encountered internal event (if any) and the event time is provided through the argument <<intermediateUpdateTime>> when calling <<fmi3CallbackIntermediateUpdate>>, along with the reason <<eventOccurred, `eventOccurred == fmi3True`>>.
+The Co-Simulation master can signal back to the FMU if and when to return early with the output arguments <<earlyReturnRequested>> and <<earlyReturnTime>> of <<fmi3CallbackIntermediateUpdate>>.
+If the FMU returns with <<earlyReturn,`earlyReturn == fmi3True`>> from <<fmi3DoStep>>, the Co-Simulation master will start event handling by calling <<fmi3EnterEventMode>> for that FMU to push the FMU into *Event Mode*.
 In this mode the Co-Simulation master is supposed to catch all events through the <<fmi3NewDiscreteStates>> function.
 
 If an early-return request of the Co-Simulation master is ignored by the FMU, then <<fmi3DoStep>> returns with <<earlyReturn,`earlyReturn == fmi3False`>>.
 The master can start a resynchronization of FMUs at an event time, if the <<currentCommunicationPoint>> has passed the event time, the master can roll-back the FMU and repeat the step with a suitable <<communicationStepSize>> (if the FMU supports the roll-back).
 
-In *Event Mode* and only after <<fmi3DoStep>> returned with <<earlyReturn,`earlyReturn == fmi3True`>> the function <<fmi3NewDiscreteStates>> may be called.
+[source, C]
+----
+include::../headers/fmi3FunctionTypes.h[tag=NewDiscreteStates]
+----
+
+In *Event Mode* and only after <<fmi3DoStep>> returned with <<earlyReturn,`earlyReturn == fmi3True`>>, and <<fmi3EnterEventMode>> was called, the function <<fmi3NewDiscreteStates>> may be called.
 Only the following output arguments may be considered:
```

Comment:
We discussed having some stronger wording here. Perhaps:  
Only the following output arguments are defined by the FMU:  
...items...
All other output arguments from <<fmi3NewDiscreteStates>>  are undefined and have no meaning in Hybrid Co-Simulation.